ui/src/index.tsx

```diff
@@ -157,6 +157,7 @@ declare global {
 
 // Older method used for pre-IE 11 compatibility
 window.basepath = basepath
+window.dataLayer = window.dataLayer || []
```

Comment:
idk how this is supposed to work, dataLayer isn't an array in quartz

Reply:
@drdelambre [i think it is](https://github.com/influxdata/quartz/blob/1efa065cf012dcb5f90c0c0bc3bf573d873f51c3/assets/js/app.jsx#L45)

Reply:
has it changed from this?
https://github.com/influxdata/quartz/blob/d691434c6bcfd564cb54d089654cf57480261423/assets/js/client/GoogleAnalyticsClient.test.ts#L6

Reply:
just bring it up because we've had a lot of false starts with this one: https://github.com/influxdata/idpe/issues/6298 and it'd be cool to add new context on this one

Reply:
that's def newer than the array.  for sure.  im fine with either.  as long as they are the same.

Reply:
i think the `window.dataLayer = []` is a fix to prevent errors from crashing the page in the event of someone calling `GA.push(event)` before the actual GA client has been loaded

Reply:
I don't understand quartz and the UI interact here at all and I don't understand how GA works. if anyone does I would love some help.

Reply:
So we're all clear: `window.dataLayer` is a data structure used by Google Tag Manager. Appending an item to this array is how we "fire" events for GTM. We can view those events in google analytics or google tag manager. We're using this system absent another one because it's already in place and it's already paid for (i.e. the costs of using it are very low relative to adopting a new one).

Quartz and Cloud both use the same google tag manager system, and quartz has its own set of events that are fired. We are able to correlate user behavior between cloud and quartz, but it's not our (ui engineers) concern on how to do that right now. Basically you can view this as: *as long as you don't overwrite events, everything should interact fine without much conflict*

In cloud, when we deploy, we overwrite our OSS index.html file [with an idpe one](https://github.com/influxdata/idpe/blob/master/apps/chronografd/index.html#L8) which sets up GTM correctly. I love the defensiveness of the line, and having it in will  not hurt anything, but it's not necessary, since this same action is done in the index.html file.


https://developers.google.com/tag-manager/devguide

Reply:
for future reference and in the spirit of sharing knowledge, we haven't used that file since  november-ish? please refer all changes to GTM in https://github.com/influxdata/monitor-ci/blob/master/deploy/Jenkinsfile.deploy#L386